hydra-test-utils/src/Test/Hydra/Prelude.hs

```diff
@@ -137,3 +156,83 @@ newtype ReasonablySized a = ReasonablySized a
 
 instance Arbitrary a => Arbitrary (ReasonablySized a) where
   arbitrary = ReasonablySized <$> reasonablySized arbitrary
+
+-- Like 'coverTable', but construct the weight requirements generically from the
+-- label type-representation.
+--
+--     data MyCoverLabel = Case1 | Case2 deriving (Generic, Data, Show)
+--
+--     forAll arbitrary $ \(lbl :: MyCoverLabel, arg) ->
+--         myProp arg
+--         & genericCoverTable @UniformCoverage [lbl]
```

Comment:
Wouldn't we want to have `genericCoverTable arg` which of course requires `Generic` for the `arg` type, but it would be more concise IMO.

Reply:
Not exactly no. We still want / need some form of labels which are distinct from the arbitrary arg, otherwise we will end up labeling ALL generated values without any kind of grouping and this is generally not what we want.

Reply:
What we _could_ do is to only use the constructor name as a label (and not look at the fields) and make sure that `arg` is a sum-type which wrapped the label groups we care about. In this case, something like:

```hs
data CloseMutation
  = MutateSignatureButNotSnapshotNumber Mutation
  | MutateSnapshotNumberButNotSignature Mutation
  | MutateSnapshotToIllFormedValue Mutation
  | MutateParties Mutation
```

would work nicely.

Reply:
Okay, this should be fixed in 8ad9a06